test_runner/batch_others/test_proxy.py

```diff
@@ -22,6 +29,154 @@ def test_password_hack(static_proxy):
         static_proxy.safe_psql('select 1', sslsni=0, user=user, password=magic)
 
 
+async def get_session_id_from_welcome_message(local_link_proxy, proc):
+    # Read line by line output of psql.
+    # We expect to see result in first 3 lines. [this is flexible, change if need be]
+    # Expected output:
+    #   A notice that contains a welcome to neon message that contains a http url in the form of
+    #   "http://dummy-uri/<psql_session_id>"
+    #   (here dummy-uri was passed from the neon_fixture that created the link_proxy
+    #   The url has to be on its own line.
+    #   Example:
+    #   "
+    #   NOTICE:  Welcome to Neon!
+    #   Authenticate by visiting:
+    #      http://dummy-uri/<psql_session_id>
+    #   "
+
+    psql_session_id = None
+
+    max_num_lines_of_welcome_message = 15
+    attempt = 0
+    log.info(f"Neon Welcome Message:")
+    for attempt in range(max_num_lines_of_welcome_message):
+        raw_line = await proc.stderr.readline()
+        if not raw_line:
+            # output ended
+            break
+        log.info(f"{raw_line}")
+        line = raw_line.decode("utf-8").strip()
+        if line.startswith("http"):
```

Comment:
This will work but it's more robust to look for the line with `dummy-uri`. That's the string you are in control of, no matter how the message changes, the link will always contain that string